### DISCUSSION TITLE
including new path for IntelliSense

### DISCUSSION THREAD

**Discussion by Benedito821** [microsoft/vscode-cpptools#7854] [Q&A]
Hello! I am trying to compile a project but I get the error that one file was not found. I added the include path in .json file but the error persists. When I run the C/C++: Log Diagnostics command I see that the path is not being added. 
Here is the .json file:
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**",
                **"/home/.../.../OTA_update_STM32_using_ESP32/components/stm_flash/include"**
            ],
            "defines": [],
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "gnu17",
            "cppStandard": "gnu++14",
            "intelliSenseMode": "linux-gcc-x64",
            "compileCommands": "${workspaceFolder}/build/compile_commands.json"
        }
    ],
    "version": 4
}

Here is a part of the diagnostic output:

Browse Paths from compile_commands.json, from workspace folder: /home/.../.../OTA_update_STM32_using_ESP32/file_serving_stm
    /home/.../.../OTA_update_STM32_using_ESP32/file_serving_stm/build
    /home/.../.../OTA_update_STM32_using_ESP32/file_serving_stm/build/config
    /home/.../.../OTA_update_STM32_using_ESP32/file_serving_stm/main

there is no /home/.../.../OTA_update_STM32_using_ESP32/components/stm_flash/include  here

any tips?

**Comment by djairao** [comment]
Is this an error that you get when compiling or is it with IntelliSense?

**Reply by Benedito821** [reply]
When I compile

**Reply by sean-mcmanus** [reply]
How are you compiling? c_cpp_properties.json doesn't control any compiling. That is done via tasks.json or another extension that handles building. Also, when you have compileCommands set, the includePath won't be used for source files that are specified in the compile_commands.json.

**Reply by Benedito821** [reply]
I am compiling using _idf.py build_. I mentioned c_cpp_properties.json because I was trying to add there the missing paths.

**Reply by sean-mcmanus** [reply]
I don't know what idf.py build is or how it's configured. Our extension or c_cpp_properties.json should have no effect on that build script.

**Reply by Benedito821** [reply]
And how can I solve the problem of file not found? The file exists on my pc. How can I include it?

**Reply by sean-mcmanus** [reply]
Your problem appears to be a build issue. I don't know how you're building. Normally, you'd need to add some -I or some flag to add a directory to your include path for building.
